lib/active_merchant/billing/gateways/pay_trace.rb

```diff
@@ -0,0 +1,259 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayTraceGateway < Gateway
+      self.test_url = 'https://api.paytrace.com'
+      self.live_url = 'https://api.paytrace.com'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://paytrace.com/'
+      self.display_name = 'PayTrace'
+
+      # Response codes based on API Response Codes found here: https://developers.paytrace.com/support/home#14000041297
+      STANDARD_ERROR_CODE_MAPPING = {
+        '1'   => STANDARD_ERROR_CODE[:error_occurred],
+        '102' => STANDARD_ERROR_CODE[:declined],
+        '103' => STANDARD_ERROR_CODE[:auto_voided],
+        '107' => STANDARD_ERROR_CODE[:unsuccessful_refund],
+        '108' => STANDARD_ERROR_CODE[:test_refund],
+        '110' => STANDARD_ERROR_CODE[:unsuccessful_void],
+        '113' => STANDARD_ERROR_CODE[:unsuccessful_capture]
+      }
+
+      def initialize(options = {})
+        requires!(options, :username, :password, :integrator_id)
+        super
+        acquire_access_token
+      end
+
+      def purchase(money, payment_or_customerid, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        if payment_or_customerid.class == String
+          post[:customer_id] = payment_or_customerid
+
+          response = commit('transactions/sale/by_customer', post)
+          check_token_response(response, 'transactions/sale/by_customer', post, options)
```

Comment:
I don't necessarily know that we should, but I'm noodling on whether it would clarify things for future generations to factor these endpoint strings out into a hash or something, just to be able to see them all in one place? e.g., 

```
ENDPOINTS = {
  customer_sale: 'transactions/sale/by_customer', 
  keyed_sale: 'transactions/sale/keyed', 
  ... 
}
```

and then calling, e.g., `check_token_response(response, ENDPOINTS[:customer_sale], post, options)`

Reply:
Love this idea! Will build it out before I submit for re-review